src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -78,6 +92,20 @@ public HubConnection(IConnection connection, IHubProtocol protocol, ILoggerFacto
             }
         }
 
+        private void TimeoutElapsed()
+        {
+            _connection.AbortAsync(new TimeoutException("Server timeout elapsed without receiving a message from the server."));
```

Comment:
Would it be useful to call out what the timeout was in the exception message?

Reply:
Most other systems don't. We could, I suppose.

Reply:
A lot of those are simply converting error codes to strings though so they don't have the timeout value :). I should be able to add that in easily.

Reply:
If this is not common elsewhere, let's not complicate this more then. :)

Reply:
Meh, it's easy and done. It'd be more work to remove it now :P